### PR TITLE
added Aave test

### PR DESCRIPTION
**Pull request by alex-sumner**

### CODE REVIEW

File: test/compoundDraft.ts [hunk 1]
```diff
@@ -1,55 +1,58 @@
...
-
-      //transfer SUPPLY_AMOUNT USDC to deployer
-      let tx = await this.USDC.connect(this.makerSigner).transfer(this.deployer.address, SUPPLY_AMOUNT);
-      await tx.wait();
-
-      //INTEGRITY CHECK
-      this.startBalance = await this.USDC.balanceOf(this.deployer.address);
-      expect(this.startBalance).to.be.not.eq(ethers.constants.Zero);
-      console.log("DEPLOYER USDC balance=", ethers.utils.formatUnits(this.startBalance, 6));
+    const UINT256_MAX = ethers.BigNumber.from("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
```
Comment (kshilov) [inline test/compoundDraft.ts:24]: Use constants please: ethers.constants.MaxUint256

details here: https://docs.ethers.io/v5/api/utils/constants/

File: test/compoundDraft.ts [hunk 2]
```diff
@@ -70,13 +73,49 @@ describe("Compound APY", function () {
...
+        await tx.wait();
+
+        let block = await ethers.provider.getBlock(tx.blockNumber);
+        let startTimestamp = block.timestamp;
+
+        console.log("...waiting for 30 seconds");
+        await delay(30000);
+        console.log("...redeem start");
+
+        tx = await this.aaveLendingPool.withdraw(this.USDC.address, UINT256_MAX, this.deployer.address);
```
Comment (kshilov) [inline test/compoundDraft.ts:104]: How is that possible? You withdraw more than you supplied ?

File: test/compoundDraft.ts [hunk 3]
```diff
@@ -70,13 +73,49 @@ describe("Compound APY", function () {
...
+
+        let diff = endTimestamp - startTimestamp;
+        let periods = (365 * 24 * 60 * 60) / diff;
 
         let endBalance = await this.USDC.balanceOf(this.deployer.address);
         console.log("DEPLOYER USDC after redeem balance=", ethers.utils.formatUnits(endBalance, 6));
 
         let profit = endBalance - this.startBalance;
-        console.log("PROFIT earned" + profit.toString() + " passed periods=" + periods);
+        console.log("PROFIT earned=" + profit.toString() + " passed periods=" + periods);
```
Comment (kshilov) [inline test/compoundDraft.ts:118]: We also should add yield farming somewhere here. It's AAVE token or something like that.

So you supply USDC, and get back profit NOT only in USDC but also in "platform token" (AAVE) - we need to calculate this profit as well.

**Review by kshilov — CHANGES_REQUESTED** [review]

File: test/compoundDraft.ts (hunk 1 above)
Comment (alex-sumner) [reply test/compoundDraft.ts:24]: No problem

File: test/compoundDraft.ts (hunk 3 above)
Comment (alex-sumner) [reply test/compoundDraft.ts:118]: Yes, that is to come...

File: test/compoundDraft.ts (hunk 2 above)
Comment (alex-sumner) [reply test/compoundDraft.ts:104]: See https://docs.aave.com/developers/the-core-protocol/lendingpool#withdraw

**Review by kshilov — APPROVED** [review]
lgtm

**Pull request merged**